src/Symfony/Component/Validator/Constraints/Collection.php

```diff
@@ -41,9 +41,12 @@ class Collection extends Composite
      */
     public function __construct($fields = null, array $groups = null, $payload = null, bool $allowExtraFields = null, bool $allowMissingFields = null, string $extraFieldsMessage = null, string $missingFieldsMessage = null)
     {
-        // no known options set? $fields is the fields array
-        if (\is_array($fields)
-            && !array_intersect(array_keys($fields), ['groups', 'fields', 'allowExtraFields', 'allowMissingFields', 'extraFieldsMessage', 'missingFieldsMessage'])) {
+        $knowOptions = array_map(function (\ReflectionParameter $parameter) {
+            return $parameter->name;
+        }, (new \ReflectionMethod(__METHOD__))->getParameters());
```

Comment:
```suggestion
        static $knowOptions;
        $knownOptions ??= array_column((new \ReflectionMethod(__METHOD__))->getParameters(), 'name');
```

Reply:
Done, I had to adjust the code for PHP 7.2. I'm wondering, is the static variable necessary since this all happens in the constructor?